libraries/chain/steem_evaluator.cpp

```diff
@@ -2307,10 +2307,11 @@ void delegate_vesting_shares_evaluator::do_apply( const delegate_vesting_shares_
    // Else the delegation is decreasing
    else /* delegation->vesting_shares > op.vesting_shares */
    {
-      FC_ASSERT( delegation->vesting_shares - op.vesting_shares >= min_delegation || op.vesting_shares.amount == 0, "Delegation must be removed or leave minimum delegation amount of ${v}", ("v", min_delegation) );
-
       auto delta = delegation->vesting_shares - op.vesting_shares;
 
+      FC_ASSERT( delta >= min_update, "Steem Power increase is not enough of a different. min_update: ${min}", ("min", min_update) );
```

Comment:
There are edge cases that `op.vesting_shares == 0` is true but `delta` be less than `min_update`, for example when account registration fee is increased. So in #925 I suggested `op.vesting_shares == 0` OR other checks.